rescan.go

```diff
@@ -0,0 +1,122 @@
+package dcrlibwallet
+
+import (
+	"math"
+	"time"
+
+	"github.com/decred/dcrwallet/errors"
+	w "github.com/decred/dcrwallet/wallet/v3"
+)
+
+func (mw *MultiWallet) RescanBlocks(walletID int) error {
+
+	wallet := mw.WalletWithID(walletID)
+	if wallet == nil {
+		return errors.E(ErrNotExist)
+	}
+
+	netBackend, err := wallet.internal.NetworkBackend()
+	if err != nil {
+		return errors.E(ErrNotConnected)
+	}
+
+	if mw.IsRescanning() || !mw.IsSynced() {
+		return errors.E(ErrInvalid)
+	}
+
+	go func() {
+		defer func() {
+			mw.syncData.mu.Lock()
+			mw.syncData.rescanning = false
+			mw.syncData.cancelRescan = nil
+			mw.syncData.mu.Unlock()
+		}()
+
+		mw.syncData.mu.Lock()
+
+		mw.syncData.rescanning = true
+
+		if mw.blocksRescanProgressListener != nil {
+			mw.blocksRescanProgressListener.OnBlocksRescanStarted(walletID)
+		}
```

Comment:
Let's move this to after unlocking the mutex to prevent potentially holding the lock for a while. I believe since the listener is not being invoked in a background thread, this would block until the listener completes responding to the notification.